### ISSUE TITLE
Missing attachment when email is not multipart and only contains an attachment 

### ISSUE THREAD

**Issue by arthurzenika** [postalserver/postal#2727]
## Describe the bug

When an email has no text or html content and is not multipart and **only** contains an attachment, postal doesn't show the attachment in the webUI. 

## To Reproduce

1. Send the attached email
2. Click on incoming messages, choose the email
3. Click on Attachments
4. See "There are no attachments for this message."

## Expected behaviour

For the attachment to be displayed and downloadable in the webUI

## Environment details 

 - OS: n/a
 - Browser n/a
 - Version : 2.1.4
 - Type : n/a

## Additional information/context

[postal_example.eml.gz](https://github.com/postalserver/postal/files/13725772/postal_example.eml.gz)

**Comment by arthurzenika** [comment]
Oh and for some context at first we though the bug was https://github.com/postalserver/postal/issues/1624

**Comment by willpower232** [comment]
I presume you're sending this message with SMTP and not the API? And also that the message arrives at its destination okay with the attachment intact?

**Comment by arthurzenika** [comment]
Indeed the message is an incoming email into postal that should then trigger an HTTP endpoint that forwards the mails to our application. To the best of my knowledge, the message does not include the attachment in the HTTP endpoint json call. 

(oh and happy new year by the way !)

**Comment by willpower232** [comment]
Haha happy new year indeed!

Whatever sends the message with Amazon SES to Postal doesn't construct it with parts, Postal won't identify the attachment at all. Even the scanners I've seen use message parts, are you constructing this message yourself or do you have another device involved?

**Comment by arthurzenika** [comment]
I was indeed surprised to see the absence of parts in the email. The email is generated by a company that sends it to our company, we're looking into the possibility of ask them to change the way it is generated. But some mail clients seems to think the email is valid, I haven't found a tool that says the email is invalid.

**Comment by willpower232** [comment]
Yeah its definitely Postal being very strict in this context but I'm not sure how extensive the changes would need to be in order to parse it.

I can't immediately see where the attachments are parsed out in either [lib/postal/smtp_server/client.rb](https://github.com/postalserver/postal/blob/main/lib/postal/smtp_server/client.rb) or [lib/postal/message_db/database.rb](https://github.com/postalserver/postal/blob/main/lib/postal/message_db/database.rb) or [lib/postal/message_db/message.rb](https://github.com/postalserver/postal/blob/main/lib/postal/message_db/message.rb)

**Comment by catphish** [comment]
This is definitely a bug. I never thought to look for attachments in a non-multipart email. I will see if we can fix it.

**Comment by deefdragon** [comment]
As a note, Ive encountered this before from gmail DMARC report emails. They also only send the attachment with the report, and no body text/html. I was hoping to use postal to assist in parsing and reporting of these DMARC emails, as opposed to grabbing the data by hand, but with this bug, its not yet possible.

**Comment by Laaknor** [comment]
I've been testing with DMARC-reports the last few days, and it was working, until today, when it suddently stoped working for mimecast.org and google.com. 

In postal headers, it says "attachment; filename=google.com!<redacted>!1719619200!1719705599.zip", and if I download the mail, it has an attachment, but when it reaches the HTTP-endpoint, it does not have an attachment, but the size of the request indicates there is an attachment there.

![image](https://github.com/postalserver/postal/assets/1952713/a9959f62-deb0-4199-87cc-26f08eece4cc)
![image](https://github.com/postalserver/postal/assets/1952713/1edd1c39-b367-4512-9577-e26eb240fc62)
![image](https://github.com/postalserver/postal/assets/1952713/09235a7c-0998-405a-ab21-8a3ab00aae8f)

**Comment by sinnbeck** [comment]
@Laaknor I just ran into the exact same issue. Did you manage to find a workaround ?

**Comment by Laaknor** [comment]
> @Laaknor I just ran into the exact same issue. Did you manage to find a workaround ?

No, I have temporary changed my DMARC-handling to another solution with a shared mailbox.

**Comment by arthurzenika** [comment]
@sinnbeck our work around was to ask for the sender to change the message, so not really, sorry.

**Comment by sinnbeck** [comment]
@Laaknor Ah too bad. I might try setting it up with mailgun instead then. At least until a fix is made down the line :) Thanks!

@arthurzenika Sadly it is dmarc emails from google and has the issue, so I don't think I can ask them to fix it :(

**Comment by deefdragon** [comment]
This is one of the bugs I have offered to sponsor getting fixed, but I have received no communication on actually doing so after the offer in the discord.  I'm also not sure the best way to go about it.

**Comment by sinnbeck** [comment]
@deefdragon Sad to hear that. I wish I had the understanding to make a pr myself, but sadly I know nothing about mail parsing :/

**Comment by sinnbeck** [comment]
I want to see if I can find a workaround for this, but I am unable to recreate the issue when sending an email myself. Does anyone know of a way to trigger the bug from a mail client (outlook, gmail etc) ?

**Comment by sinnbeck** [comment]
I believe I have found a work around. Instead of letting postal parse the email, I just send the raw message to my server as form data, and then use a mime parser there to extract the attachment. In testing it worked like a charm. I will report back if there are any issues down the line :)

**Comment by sinnbeck** [comment]
I have been running it for 4 days now and everything seems to be working as expected :) 30 reports so far and several of them from google.

**Comment by Killianzrca** [comment]
Hello,
I'm also experiencing this problem, the problem stemming from an email sent from a hard device.
The email has no body and the error returned is `FrozenError: can't modify frozen String: ""
`
